docs/usage/user.asciidoc

```diff
@@ -0,0 +1,396 @@
+
+In many cases, events that only mention one user should populate the user fields
+at the root of the event, even if the user is not the one performing the action.
+
+In cases where a purpose-specific user field such as `url.username` is populated,
+`user.name` should also be populated with the same user name.
+
+[source,json]
+-----------
+{
+  "url": { "username": "alice" }, <1>
+  "user": { "name": "alice" }, <2>
+  "related": { "user": ["alice"] }
+}
+-----------
+<1> Purpose-specific username field
+<2> Username copied to `user.name` to establish `user.name` as a reliable baseline.
+
+[discrete]
+[[ecs-user-usage-remote-logons]]
+====== Remote Logons
+
+When users are crossing host boundaries, the users are captured at
+`source.user` and `destination.user`.
+
+Examples of data sources where this is applicable:
+
+* Remote logons via ssh, kerberos
+* Firewalls observing network traffic
+
+In order to align with ECS' design of having `user` at the root of the event as the
+user performing the action, all `source.user` fields should be copied to `user` at the root.
+
+Here's an example where user "alice" logs on to another host as user "deus":
+
+[source,json]
+-----------
+{
+  "user": {
+    "name": "alice"
+  },
+  "source": {
+    "user": {
+      "name": "alice"
+    },
+    "ip": "10.42.42.42"
+  },
+  "destination": {
+    "user": {
+      "name": "deus"
+    },
+    "ip": "10.42.42.43"
+  },
+  "related": { "user": ["alice", "deus"] }
+}
+-----------
+
+Whenever an event source populates the `client` and `server` fields in addition
+to `source` and `destination`, the user fields should be copied accordingly as well.
+
+[discrete]
+[[ecs-user-usage-privilege-changes]]
+====== Privilege Changes
+
+The `user.effective` fields are relevant when there's a privilege escalation or demotion
+and it's possible to determine the user requesting/performing the escalation.
+
+Use the `user` fields at the root to capture who is requesting the privilege change,
+and `user.effective` to capture the requested privilege level, whether or not the
+privilege change was successful.
+
+Here are examples where this is applicable:
+
+* A user changing identity on a host.
+** Examples: sudo, su, Run as.
+* Running a program as a different user. Examples:
+** A trusted user runs a specific admin command as root via a mechanism such as the Posix setuid/setgid.
+** A service manager with administrator privileges starts child processes as limited
+    users, for security purposes (e.g. root runs Apache HTTPD as user "apache")
+
+In cases where the event source only gives information about the effective user
+and not who requested different privileges, the `user` fields at the root of the
+event should be used instead.
+
+Here's an example of user "alice" running a command as root via sudo:
+
+[source,json]
+-----------
+{
+  "user": {
+    "name": "alice",
+    "id": "1001",
+    "effective": {
+      "name": "root",
+      "id": "1"
```

Comment:
Not related to the review at all, but seeing `root` and `id` value of 1 got me off down the rabbit hole.

It has absolutely no bearing; I'm just hardwired to expect a UID/GID of zero. 😂

Reply:
Oops yeah, the idea isn't to troll anyone with this 😂  Got confused with PID 1, here I think. I'll fix